third_party/nvfuser/csrc/fusion.h

```diff
@@ -235,6 +235,20 @@ class TORCH_CUDA_CU_API Fusion : public IrContainer {
     return io_alias_;
   }
 
+  using LoopRotationParam = std::vector<
+      std::tuple<TensorView*, int64_t, std::unordered_set<Statement*>>>;
+
+  const LoopRotationParam& getLoopRotationParam() const {
+    return loop_rotation_param_;
+  }
+
+  void rotateLoop(
```

Comment:
Question on interface. This seems inconsistent compared to the rest of the API. For example, double buffering, which is kind of a similar concept, is done as: `tv->doubleBuffer();`. For loop rotation, why not `loop_tv->rotate(axis, selection)`?

Reply:
For double buffer, the scope is just limited to this tensor "I want to double buffer this tensor", but for loop rotation, the scope is an entire loop with all exprs inside the loop. So I belive it would be better to put the interface to more global places like fusion. Also, putting it to fusion is easier to access during lowering, because it is already there, and we don't need to write a pass to collect that information.

Reply:
Hmm, I don't agree. It's a property on a loop, and a loop is represented by `IterDomain`. So, something like `loop_tv->axis(axis)->rotate(selection)` should also be enough. I don't understand why it needs to be a global property.

Reply:
Will this cause unterminated recursion in ir cloner?
```
loop_tv->id->selection->loop_tv->id->.....
```

Reply:
Looks like generally we should not make a tv directly or indirectly contain a pointer to tv, this could potentially cause unterminated recursion. `computeWith` is an exception, because we know that `computeWith` contains only consumer, which is guaranteed to never have a cyclic containing relationship.

Reply:
I don't consider the infomation "XXX loop need to rotate XXX exprs" as an attribute of a tensor or an ID. I more consider it as a lowering hint/requirement of the kernel. And I don't want the selection to be limited to tensors. I want to be more flexible, and using that same interface, we can also select exprs, instead of tensors.

And because fusion schedulers don't have access to kernel structure, I generally don't think it makes sense to let fusion schedulers to provide loop rotation information. For example, currently, if I select the double buffer load for rotation, it will not only rotate the double buffer main loop, but it will also rotate the double buffer prologue loop. And the current interface does not allow control of that. The only reason I designed this interface is to make it convenient for writing unit tests. I don't expect this interface to be a utility like `computeAt` which will be used widely in schedulers. I consider this interface as a lightweight hacking point for unit tests and matmul scheduler to talk to lowering, so I don't want it to be buried deep in our design. And I do think we need to revisit the design of interface when we get another user that is not a matmul or unit test.

Reply:
Why tensors are not enough? Why do you want to be able to specify exprs? I'm thinking about what the interface should be for a fusion. 

For the case like double buffering and loop rotation, I see that you may want to select a particular expr among all exprs defining the same tensor, i.e., the expr appearing in the main loop but not in the prologue. But that's not a `Fusion` anymore. It seems it only makes sense in the lowering phase.

Reply:
OK, fair enough. The only remaining comment regarding this is https://github.com/csarofeen/pytorch/pull/2500#discussion_r1119490970